packages/runtime/src/controls/link.ts

```diff
@@ -0,0 +1,44 @@
+type OpenPageLink = {
+  type: 'OPEN_PAGE'
+  payload: { pageId: string | null | undefined; openInNewTab: boolean }
+}
+
+type OpenURLLink = { type: 'OPEN_URL'; payload: { url: string; openInNewTab: boolean } }
+
+type SendEmailLink = {
+  type: 'SEND_EMAIL'
+  payload: { to: string; subject?: string; body?: string }
+}
+
+type CallPhoneLink = { type: 'CALL_PHONE'; payload: { phoneNumber: string } }
+
+type ScrollToElementLink = {
+  type: 'SCROLL_TO_ELEMENT'
+  payload: {
+    elementIdConfig: { elementKey: string; propName: string } | null | undefined
+    block: 'start' | 'center' | 'end'
+  }
+}
+
+export type LinkControlData =
+  | OpenPageLink
+  | OpenURLLink
+  | SendEmailLink
+  | CallPhoneLink
+  | ScrollToElementLink
+
+export const LinkControlType = 'makeswift::controls::link'
+
+type LinkControlConfig = {
+  label?: string
+  defaultValue?: LinkControlData
```

Comment:
I'm not sure we want to expose the default value just yet since it's an internal data structure that we probably don't want people typing themselves. For example, for the `Color` control, I exposed `defaultValue` because I was able to make it any hex or CSS color without requiring the user to provide a responsive swatch ID. I think the default value here should probably be more like the "mapped" value we return or also just support a simple URL (i.e., `/about`).

I suggest we remove this option for now and we can try and figure out a better API for the default value later.

Reply:
Yeah I agree. Updated.